Framework/Algorithms/src/CopySample.cpp

```diff
@@ -120,22 +125,25 @@ void CopySample::exec() {
                         << (outMDWS->getNumExperimentInfo() - 1) << "). Will use sample number 0 instead\n";
         outputSampleNumber = 0;
       }
-      copyParameters(sample, outMDWS->getExperimentInfo(static_cast<uint16_t>(outputSampleNumber))->mutableSample(),
-                     copyName, copyMaterial, copyEnvironment, copyShape, copyLattice, copyOrientation);
+      copyParameters(
+          sample, outMDWS->getExperimentInfo(static_cast<uint16_t>(outputSampleNumber))->mutableSample(), copyName,
+          copyMaterial, copyEnvironment, copyShape, copyLattice, copyOrientation,
+          outMDWS->getExperimentInfo(static_cast<uint16_t>(outputSampleNumber))->run().getGoniometer().getR());
```

Comment:
Similar to above, I suggest you do `auto ei = outMDWS->getExperimentInfo(static_cast<uint16_t>(outputSampleNumber))` before the call to copyParameters